magento/partners-magento2b2b#370: determine mftf coverage in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules

dhaecker opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/420
https://github.com/magento/partners-magento2b2b/pull/496
<!-- related pull request placeholder -->

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. magento/partners-magento2b2b#370: [Determine Coverage]: Add Meaningful Titles and Descriptions For All MFTF Tests in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules
1. https://jira.corp.magento.com/browse/MC-39333:  Grids Frequently (but randomly) Fail to Load & Throw getData Undefined Error in Console

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. ...
2. ...

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Approval Ticket
https://jira.corp.magento.com/browse/MC-39488

dhaecker committed:
magento/partners-magento2b2b#370: [Determine Coverage]: Add Meaningful Titles and Descriptions For All MFTF Tests in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules

- Fixing MFTF annotations for Company module
- Fixing AdminCheckThatReorderIsNotAvailableForTeamTest

dhaecker committed:
Merge branch 'magento2/2.4-develop' into 370_determine_mftf_coverage

dhaecker committed:
magento/partners-magento2b2b#370: [Determine Coverage]: Add Meaningful Titles and Descriptions For All MFTF Tests in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules

 - Fixing MFTF annotations for Company module
 - Cleaning up AdminReplaceCompanyAdminTest
 - Cleaning up AdminSetActiveCustomerGridWorkCorrectTest

dhaecker committed:
Merge branch 'magento2/2.4-develop' into 370_determine_mftf_coverage

dhaecker committed:
Merge branch 'magento2/2.4-develop' into 370_determine_mftf_coverage

dhaecker committed:
Merge branch 'magento2/2.4-develop' into 370_determine_mftf_coverage

dhaecker committed:
magento/partners-magento2b2b#370: [Determine Coverage]: Add Meaningful Titles and Descriptions For All MFTF Tests in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules

 - Updating AdminSetActiveCustomerGridWorkCorrectTest

dhaecker committed:
magento/partners-magento2b2b#370: [Determine Coverage]: Add Meaningful Titles and Descriptions For All MFTF Tests in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules

 - Cleaning up StorefrontNonCompanyCustomerCreatesCompanyTest
 - Cleaning up StorefrontCompanyUserCreateTest
 - Cleaning up AdminVerifyCustomerGroupOfACustomerChangesWhenACompanyIsAssignedToTheCustomerTest
 - Cleaning up AdminVerifyCustomerGroupOfCompanyCustomerChangesWhenNewCustomerGroupIsAssignedToACompanyTest

dhaecker committed:
magento/partners-magento2b2b#370: [Determine Coverage]: Add Meaningful Titles and Descriptions For All MFTF Tests in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules

 - Cleaning up StorefrontVerifyOrdersVisibleInCompanyStructureAcrossMultipleWebsitesTest
 - Adding inline comments to before & after sections

lbajsarowicz commented on app/code/Magento/Checkout/Test/Mftf/ActionGroup/StorefrontClickOrderLinkFromCheckoutSuccessPageActionGroup.xml:
```diff
@@ -0,0 +1,19 @@
+        </annotations>
+        <click selector="{{CheckoutSuccessMainSection.orderLink}}" stepKey="clickOrderLink"/>
+        <waitForPageLoad stepKey="waitForPageLoad"/>
+        <waitForText selector="{{StorefrontCustomerAccountMainSection.pageTitle}}" userInput="Order # " stepKey="waitForPageTitle"/>
```
```suggestion
        <waitForText selector="{{StorefrontCustomerAccountMainSection.pageTitle}}" userInput="Order #" stepKey="waitForPageTitle"/>
```

dhaecker replied:
done

lbajsarowicz commented on app/code/Magento/Customer/Test/Mftf/ActionGroup/AdminSetCustomerActiveViaGridActionGroup.xml:
```diff
@@ -0,0 +1,26 @@
+        <click selector="{{AdminCustomerGridMainActionsSection.actions}}" stepKey="clickActions"/>
+        <waitForElementVisible selector="{{AdminCustomerGridMainActionsSection.setActive}}" stepKey="waitForDropDownOpen"/>
+        <click selector="{{AdminCustomerGridMainActionsSection.setActive}}" stepKey="clickSetActive"/>
+        <waitForPageLoad stepKey="waitForLoad"/>
+        <waitForText selector="{{AdminMessagesSection.success}}" userInput="A total of 1 record(s) were updated." stepKey="seeSuccessMessage"/>
```
Could you use existing `Assert` Action Group in Test, as in some cases we might like to assert that the action fails?

dhaecker replied:
@lbajsarowicz are you suggesting that i should remove the `waitForText` line from this actiongroup & use an assert actiongroup after i reference `AdminSetCustomerActiveViaGridActionGroup` in the test?

my assumption was that this actiongroup i wrote could be for the positive scenario & if someone needs to specifically test the negative case, then they could extend this actiongroup & merge with the `waitForText` action 🤷 

what are your thoughts between the two approaches?

lbajsarowicz replied:
You can rename the Action Group then to include state assertion `Successfully`.

dhaecker replied:
done

lbajsarowicz commented on app/code/Magento/Customer/Test/Mftf/ActionGroup/StorefrontGoToCustomerAddressesPageActionGroup.xml:
```diff
@@ -0,0 +1,17 @@
+        <annotations>
+            <description>Goes to the storefront customer addresses page</description>
+        </annotations>
+        <amOnPage url="{{StorefrontCustomerAddressesPage.url}}" stepKey="goToAddressPage"/>
+        <waitForPageLoad stepKey="waitForPageLoad"/>
```
`<amOnPage` already includes `waitForPageLoad` in its implementation, you don't need to add it manually.

dhaecker replied:
done

lbajsarowicz reviewed (changes requested):
🟡  I have some minor request for you, regarding Action Group reusability. Really appreciate your input!

dhaecker committed:
magento/partners-magento2b2b#370: [Determine Coverage]: Add Meaningful Titles and Descriptions For All MFTF Tests in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules

 - Fixing PR feedback

dhaecker committed:
Merge branch '2.4-develop' into 370_determine_mftf_coverage

lbajsarowicz commented:
@magento run all tests

dhaecker committed:
magento/partners-magento2b2b#370: [Determine Coverage]: Add Meaningful Titles and Descriptions For All MFTF Tests in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules

 - Fixing StorefrontCreateAnOrderWithCompanyCreditAndPurchaseOrderNumberTest
 - Cleaning up StorefrontCreateAnOrderWithCompanyCreditTest
 - Cleaning up StorefrontCreateInvoiceForPendingOrderTest

dhaecker committed:
magento/partners-magento2b2b#370: [Determine Coverage]: Add Meaningful Titles and Descriptions For All MFTF Tests in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules

 - Cleaning up StorefrontVerifyCompanyCreditStaticsForAnOrderWithOneRefundedProductTest
 - Cleaning up StorefrontVerifyCompanyCreditStaticsForAnOrderWithReimburseBalanceTest
 - Cleaning up StorefrontVerifyCreditAndCommentHistoryForCanceledOrderTest
 - Deprecating unused entities

dhaecker committed:
Merge branch '2.4-develop' into 370_determine_mftf_coverage

dhaecker committed:
Merge branch '2.4-develop' into 370_determine_mftf_coverage

dhaecker committed:
magento/partners-magento2b2b#370: [Determine Coverage]: Add Meaningful Titles and Descriptions For All MFTF Tests in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules

 - Fixing failing tests

dhaecker committed:
magento/partners-magento2b2b#370: [Determine Coverage]: Add Meaningful Titles and Descriptions For All MFTF Tests in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules

 - Fixing failing/flaky tests

danmooney2 commented on app/code/Magento/Customer/Test/Mftf/ActionGroup/StorefrontCustomStoreCustomerLogoutActionGroup.xml:
```diff
@@ -0,0 +1,20 @@
+            <description>Goes to the storefront Customer Logout page for a custom store. Must have Add Store Code To Urls enabled.</description>
+        </annotations>
+        <arguments>
+            <argument name="storeCode"/>
```
type="string" would help?

dhaecker replied:
yes good call. will update

danmooney2 commented on app/code/Magento/Customer/Test/Mftf/ActionGroup/StorefrontCustomStoreLoginActionGroup.xml:
```diff
@@ -0,0 +1,20 @@
+            <description>Goes to the storefront Customer Sign In page for a custom store. Logs in using the provided Customer. Must have Add Store Code To Urls enabled</description>
+        </annotations>
+        <arguments>
+            <argument name="storeCode"/>
```
type="string" would help?

danmooney2 commented on app/code/Magento/Customer/Test/Mftf/ActionGroup/StorefrontCustomStoreNavigateToCustomerOrdersHistoryPageActionGroup.xml:
```diff
@@ -0,0 +1,19 @@
+            <description>Goes to the storefront Customer Order History page for a custom store. Must have Add Store Code To Urls enabled</description>
+        </annotations>
+        <arguments>
+            <argument name="storeCode"/>
```
type="string" would help?

danmooney2 committed:
MC-39333: Grids Frequently Fail to Load & Throw getData Undefined Error in Console

- Add Magento_Ui/js/grid/data-storage as a dependency to base grid provider component

gabrieldagama commented:
@magento create issue

dhaecker commented:
@gabrieldagama issue already exists in partners-b2b. this PR is simply supporting the b2b PR & issue

danmooney2 committed:
Merge remote-tracking branch 'origin/MC-39333' into 370_determine_mftf_coverage

dhaecker committed:
magento/partners-magento2b2b#370: [Determine Coverage]: Add Meaningful Titles and Descriptions For All MFTF Tests in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules

 - Fixing PR feedback
 - Adding equals checks to company credit actiongroups

dhaecker committed:
Merge branch '370_determine_mftf_coverage' of github.com:dhaecker/magento2 into 370_determine_mftf_coverage

dhaecker committed:
magento/partners-magento2b2b#370: [Determine Coverage]: Add Meaningful Titles and Descriptions For All MFTF Tests in the Company, CompanyCredit, CompanyPayment, CompanyShipping, & ConfigurableSharedCatalog Modules

 - Moving B2B elements from CE to B2B

dhaecker commented:
@magento run Functional Tests CE

engcom-Delta reviewed (changes requested):
Hi @dhaecker I was able to reproduce issue in https://jira.corp.magento.com/browse/MC-39333 . With PR changes this issue is fixed, but i got another error in dev console:
```
Uncaught TypeError: Cannot read property 'offsetLeft' of undefined
    at UiClass.checkTableWidth (toolbar.js:363)
    at UiClass.run (toolbar.js:237)
    at Object.<anonymous> (jquery.js:3417)
    at fire (jquery.js:3238)
    at Object.fireWith [as resolveWith] (jquery.js:3368)
    at Object.deferred.<computed> [as resolve] (jquery.js:3467)
    at invoke (resolver.js:90)
    at Array.forEach (<anonymous>)
    at resolve (resolver.js:100)
    at resolver.js:145
```
**Manual testing scenario:**
1. Set env to development mode
2. Delete Magento_Ui/js/grid/data-storage.js from pub/static (in my case I used `rm -rf pub/static/*`)
3. Add following snippet to pub/static.php above bootstrap instantiation:
```
if (strpos($_REQUEST['resource'], 'Magento_Ui/js/grid/data-storage.js') !== false) {
    sleep(8);
}
```
4. Go to any page where UI component grid loads with console open. Ensure Magento_Ui/js/grid/data-storage.js takes significant time (>8 seconds) to return a response

**Result:**
:x:  Note error in console `Cannot read property 'offsetLeft' of undefined`
![#30957Pr](https://user-images.githubusercontent.com/51681379/100334786-e5399980-2fdc-11eb-9bf8-a286f0236f64.png)


Could you take a look?

dhaecker committed:
Merge branch '2.4-develop' into 370_determine_mftf_coverage

dhaecker commented:
@magento run Functional Tests CE

dhaecker committed:
magento/magento2-page-builder#559: User sees no predefined gutters

- Fixing failing tests

dhaecker committed:
Merge branch '370_determine_mftf_coverage' of github.com:dhaecker/magento2 into 370_determine_mftf_coverage

dhaecker commented:
@magento run all tests
